src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2150,6 +2167,87 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         return std::pow(cursorDistanceFromBorder, 2.0) / 25.0 + 2.0;
     }
 
+    // Method Description:
+    // - Async handler for the "Drop" event. If a file was dropped onto our
+    //   root, we'll try to get the path of the file dropped onto us, and write
+    //   the full path of the file to our terminal connection. Like conhost, if
+    //   the path contains a space, we'll wrap the path in quotes.
+    // Arguments:
+    // - e: The DragEventArgs from the Drop event
+    // Return Value:
+    // - <none>
+    winrt::fire_and_forget TermControl::_DoDragDrop(DragEventArgs const e)
+    {
+        if (e.DataView().Contains(StandardDataFormats::StorageItems()))
+        {
+            auto items = co_await e.DataView().GetStorageItemsAsync();
+            if (items.Size() > 0)
+            {
+                for (auto item : items)
+                {
+                    const auto fullPath = item.Path();
+                    const auto containsSpaces = std::find(fullPath.begin(),
+                                                          fullPath.end(),
+                                                          L' ') != fullPath.end();
+
+                    auto lock = _terminal->LockForWriting();
+                    if (containsSpaces)
+                    {
+                        WriteInput(L"\"");
+                    }
+                    WriteInput(fullPath);
+                    if (containsSpaces)
+                    {
+                        WriteInput(L"\"");

```

Comment:
This sequence of `WriteInput`s seems like it can be preempted by input coming in off the UI thread when it's handling user input. We may want to wrap fullPath in quotes and send it all at once.